src/etc/inc/plugins.inc.d/dhcpd.inc

```diff
@@ -462,13 +462,30 @@ function dhcpd_radvd_configure($verbose = false, $blacklist = array())
 
         $dnslist = array();
 
+        /* XXX We'll skip this interface unless a valid address can be found for it. */
+
+        $hasipv6 = false; 
         list ($ifcfgipv6, $networkv6) = interfaces_primary_address6($if, $realif, $ifconfig_details);
 
+        if (is_ipaddrv6($ifcfgipv6)) {
+            $hasipv6 = true;
+         }
+
+        foreach (config_read_array('virtualip', 'vip') as $vip) {
+            if ($vip['interface'] != $if || !is_ipaddrv6($vip['subnet'])) {
+                continue;
+            } else {
+                $hasipv6 = true;
+            }
+        }
```

Comment:
interfaces_primary_address  will not emit a VIP so I think this can be removed....